### ISSUE TITLE
When using a relative moduledir r10k deploys the modules relative to the current directory

### ISSUE THREAD

**Issue by igalic** [puppetlabs/r10k#43]
```
# The location to use for storing cached Git repos
:cachedir: '/var/tmp/r10k-cache'

# A list of git repositories to create
:sources:
  :modules:
    remote: 'git@panic:puppet-modules'
    basedir: '/tmp/puppet/modules'

# This directory will be purged of any directory that doesn't map to a
# git branch
:purgedirs:
  - '/tmp/puppet/modules'
```

With Puppetfiles ala

```
moduledir 'production'

forge "http://forge.puppetlabs.com"

mod 'puppetlabs/mysql', "0.6.1"
mod 'puppetlabs/nodejs', "0.2.1"
mod 'puppetlabs/stdlib', "3.2.0"
mod 'puppetlabs/vcsrepo', "0.1.1"
mod 'ripienaar/concat', "0.2.0"
mod "ispavailability/elasticsearch", "0.0.7"

mod "composer",
:git => "git@panic:puppet-composer"

mod "git",
:git => "git@panic:puppet-git"

mod "gitpubsub",
:git => "git@panic:puppet-gitpubsub"
```

results in r10k deploying those modules into `$PWD/production/`

See this gist for a complete example: https://gist.github.com/igalic/5767142

**Comment by adrienthebo** [comment]
I believe that this is solved by GH-141.

**Comment by adrienthebo** [comment]
Verified as resolved in 4ac4f76. Note that `moduledir` must be specified before any modules, otherwise undefined behavior may result. I've GH-204 to address that shortcoming.
